packages/grafana-ui/src/components/FileDropzone/FileListItem.test.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { FileListItem } from './FileListItem';
+
+const file: any = {
+  name: 'someFile.jpg',
+  size: 1680206,
+  lastModified: 1604849095696,
+  type: 'image/jpeg',
+};
+
+describe('The FileListItem component', () => {
+  it('should show an error message when error prop is not null', () => {
+    render(<FileListItem file={{ file, id: '1', error: new DOMException('error') }} />);
+
+    expect(screen.getByText('error')).toBeInTheDocument();
+    expect(screen.queryByLabelText('Retry')).not.toBeInTheDocument();
+  });
+
+  it('should show a retry icon when error is not null and retryUpload prop is passed', () => {
+    const retryUpload = jest.fn();
+    render(<FileListItem file={{ file, id: '1', error: new DOMException('error'), retryUpload }} />);
+
+    fireEvent.click(screen.getByLabelText('Retry'));
+
+    expect(screen.getByText('error')).toBeInTheDocument();
+    expect(screen.getByLabelText('Retry'));
+    expect(retryUpload).toBeCalledTimes(1);
+  });
+
+  it('should show a progressbar when the progress prop has a value', () => {
+    render(<FileListItem file={{ file, id: '1', error: null, progress: 800000 }} />);
+
+    expect(screen.queryByText('Cancel')).not.toBeInTheDocument();
```

Comment:
```suggestion
    expect(screen.queryByRole('button', {name: 'Cancel'})).not.toBeInTheDocument();
```
Better to use `byRole` whenever possible as it's better for testing from accessibility point of view and prevents from matching some random text on screen by mistake.

Reply:
Got it, however it won't work here as I make sure that it is not in the dom this time.